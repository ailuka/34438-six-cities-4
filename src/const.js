export const EstateType = {
  APARTMENT: `apartment`,
  HOUSE: `house`,
  HOTEL: `hotel`,
  ROOM: `room`,
};

export const CityName = {
  AMSTERDAM: `Amsterdam`,
  BRUSSELS: `Brussels`,
  COLOGNE: `Cologne`,
  DUSSELDORF: `Dusseldorf`,
  HAMBURG: `Hamburg`,
  PARIS: `Paris`,
};

export const MapClass = {
  MAP_MAIN: `cities__map`,
  MAP_DETAILS: `property__map`,
};

export const OfferClassNamesForPageType = {
  main: {
    article: `cities__place-card`,
    image: `cities__image-wrapper`,
    list: `cities__places-list tabs__content`,
    info: ``,
  },
  details: {
    article: `near-places__card`,
    image: `near-places__image-wrapper`,
    list: `near-places__list`,
    info: ``,
  },
  favorites: {
    article: `favorites__card `,
    image: `favorites__image-wrapper`,
    list: ``,
    info: `favorites__card-info`,
  }
};

export const KeyCodes = {
  ENTER: 13,
  SPACEBAR: 32,
};

export const SortTypes = {
  POPULAR: 1,
  PRICE_LOW_TO_HIGH: 2,
  PRICE_HIGH_TO_LOW: 3,
  TOP_RATED_FIRST: 4,
};

export const AppRoute = {
  ROOM: `/offer/:id`,
  LOGIN: `/login`,
  FAVORITE: `/favorite`,
  ROOT: `/`,
};
