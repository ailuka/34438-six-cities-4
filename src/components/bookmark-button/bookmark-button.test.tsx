import React from "react";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import {BookmarkStyle} from "../../const.js";
import BookmarkButton from "./bookmark-button.jsx";

const mockStore = configureStore([]);
const store = mockStore({});
const offer = {id: 1};

describe(`BookmarkButton snapshot`, () => {
  it(`should render with small button and with "To bookmarks" text`, () => {
    const tree = renderer.create(
        <Provider store={store}>
          <BookmarkButton
            offerId={offer.id}
            isBookmarked={false}
            buttonStyle={BookmarkStyle.SMALL_BUTTON}
          />
        </Provider>
    )
    .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`should render with big button and with "In bookmarks" text`, () => {
    const tree = renderer.create(
        <Provider store={store}>
          <BookmarkButton
            offerId={offer.id}
            isBookmarked={true}
            buttonStyle={BookmarkStyle.BIG_BUTTON}
          />
        </Provider>
    )
    .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
