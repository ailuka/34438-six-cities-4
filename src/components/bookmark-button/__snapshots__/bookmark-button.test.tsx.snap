// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookmarkButton snapshot should render with big button and with "In bookmarks" text 1`] = `
<button
  className="property__bookmark-button property__bookmark-button--active button"
  onClick={[Function]}
  type="button"
>
  <svg
    className="place-card__bookmark-icon"
    height="33"
    width="31"
  >
    <use
      xlinkHref="#icon-bookmark"
    />
  </svg>
  <span
    className="visually-hidden"
  >
    In bookmarks
  </span>
</button>
`;

exports[`BookmarkButton snapshot should render with small button and with "To bookmarks" text 1`] = `
<button
  className="place-card__bookmark-button  button"
  onClick={[Function]}
  type="button"
>
  <svg
    className="place-card__bookmark-icon"
    height="19"
    width="18"
  >
    <use
      xlinkHref="#icon-bookmark"
    />
  </svg>
  <span
    className="visually-hidden"
  >
    To bookmarks
  </span>
</button>
`;
